presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/db/ResourceGroupsDao.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.resourceGroups.db;
+import org.skife.jdbi.v2.sqlobject.Bind;
+import org.skife.jdbi.v2.sqlobject.SqlQuery;
+import org.skife.jdbi.v2.sqlobject.SqlUpdate;
+import org.skife.jdbi.v2.sqlobject.customizers.Mapper;
+
+import java.util.List;
+
+public interface ResourceGroupsDao
+{
+    @SqlUpdate("CREATE TABLE IF NOT EXISTS resource_groups_global_properties (\n" +
+            "cpu_quota_period VARCHAR(128) NULL)")
+    void createResourceGroupsGlobalPropertiesTable();
+
+    @SqlUpdate("CREATE TABLE IF NOT EXISTS resource_groups (\n" +
```

Comment:
I think we should make a couple changes to this table. It looks like `resource_group` is the fully qualified resource group name. I don't think this is a good idea, because it leads to issues with denormalization, and implicit constraints on the table. For example, the `checkState` for `"No parent record found for %s"` in the manager.
Instead lets enforce this in the DB. You can replace `resource_group` with a `id INT` column, a `parent_id` which is a nullable reference to `id`, and a `name VARCHAR(...)` column.
Also, the selector table should reference resource groups using the `id` column, rather than the fully-qualified name